File: CS/WPFDataGridApp15/DataClass.cs
using System;
using System.Collections.ObjectModel;

namespace WPFDataGridApp15
{
	public class DataClass
	{
		private static Random rnd;
		public static Random Rnd
		{
			get
			{
				if ( rnd == null )
					rnd = new Random();

				return rnd;
			}
		}

		public DataClass(int seed)
		{
			IntValue = seed;
			Text = "Text " + seed;
			DateValue = DateTime.Now.AddDays(-seed);

			ChildData = new ObservableCollection<ChildDataClass>();
			for ( int i = 0; i < Rnd.Next(1, 10); i++ )
				ChildData.Add(new ChildDataClass() { ChildTextValue = "Detail value " + i });
		}

		public DataClass()
		{
		}

		public int IntValue { get; set; }
		public string Text { get; set; }
		public DateTime DateValue { get; set; }
		public ObservableCollection<ChildDataClass> ChildData { get; private set; }
	}

	public class ChildDataClass
	{
		public string ChildTextValue { get; set; }
	}
}

File: VB/WPFDataGridApp15/DataClass.vb
Imports System
Imports System.Collections.ObjectModel

Namespace WPFDataGridApp15

    Public Class DataClass

        Private _ChildData As ObservableCollection(Of WPFDataGridApp15.ChildDataClass)

        Private Shared rndField As Random

        Public Shared ReadOnly Property Rnd As Random
            Get
                If rndField Is Nothing Then rndField = New Random()
                Return rndField
            End Get
        End Property

        Public Sub New(ByVal seed As Integer)
            IntValue = seed
            Text = "Text " & seed
            DateValue = Date.Now.AddDays(-seed)
            ChildData = New ObservableCollection(Of ChildDataClass)()
            For i As Integer = 0 To Rnd.Next(1, 10) - 1
                ChildData.Add(New ChildDataClass() With {.ChildTextValue = "Detail value " & i})
            Next
        End Sub

        Public Sub New()
        End Sub

        Public Property IntValue As Integer

        Public Property Text As String

        Public Property DateValue As Date

        Public Property ChildData As ObservableCollection(Of ChildDataClass)
            Get
                Return _ChildData
            End Get

            Private Set(ByVal value As ObservableCollection(Of ChildDataClass))
                _ChildData = value
            End Set
        End Property
    End Class

    Public Class ChildDataClass

        Public Property ChildTextValue As String
    End Class
End Namespace
